CONTRIBUTING.md: change path of maintainer list to `CODEOWNERS` (#500)
TheAlgorithms/Go: CONTRIBUTING.md

## Contributing

### Maintainer/reviewer

> **Please check the [reviewer code](https://github.com/TheAlgorithms/Go/blob/master/DIRECTORY.md) file for maintainers and reviewers.**

Reviewer: Instead of removing this section, point it to the [`CODEOWNERS`](https://github.com/TheAlgorithms/Go/blob/master/.github/CODEOWNERS) file.

> ### Maintainer/reviewer

Reviewer: ```suggestion
### Maintainers
```

> Please check the [reviewer code](https://github.com/TheAlgorithms/Go/blob/master/.github/CODEOWNERS) file for maintainers and reviewers.

Reviewer: ```suggestion
Please check the [`CODEOWNERS`](https://github.com/TheAlgorithms/Go/blob/master/.github/CODEOWNERS) file for a list of repository maintainers.
```
